corehq/apps/data_dictionary/templates/data_dictionary/partials/confirmation_modals.html

```diff
@@ -113,7 +113,16 @@ <h4 class="modal-title">
             <div class="modal-body">
                 <p>
                     {% blocktrans %}
-                        There are currently no cases that store data with this case property, and so it is safe to delete.
+                        There are currently no cases that store data with this case property, so it is safe to delete.
+                    {% endblocktrans %}
+                </p>
+                <p>
+                    {% blocktrans %}
+                        Please note that there may be a delay of up to 10 minutes associated with this action. If a new form
+                        or case is updated to the server during this time that makes this property unsafe to delete, this will
+                        only be reflected in the Data Dictionary after the expected time. Please refer to
+                        <a href="https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143944977/Data+Dictionary" target="_blank">the documentation</a>
+                        for additional information about this action.
```

Comment:
> Please note that there may be a delay of up to 10 minutes associated with this action

This makes it sound like the delete would take 10 minutes. So, I think its a bit confusing. Additionally, we should not convey the cache time since it could change and we might updating it here.

we could say something like,
There currently seems to be no cases that store data with this case property, so it should be safe to delete.
If any form that added data on this case property for a case was recently submitted, it might take some time to process, in which case this case property will get added back to Data dictionary.

Reply:
Yeah that's a good point. How does the below wording sound?

> Please note, this case property might not be safe to delete if there were recent form or case changes submitted to the server which use this case property. In such a case, it might take some time to properly reflect in the Data Dictionary. Please refer to the documentation for additional information about this action.

I've also reached out to an AE to get their thoughts around the above wording.

Reply:
> Please note, this case property might not be safe to delete

Even though both options are saying the same thing, I think this one is a bit vague and is leaving the user a bit indecisive making it look like this could be dangerous.
I prefer the text shared before, because its stating the facts as it is and then telling the user that its not an issue even if they go ahead with it.

Definitely good to check with AEs to understand from user's perspective.

Reply:
Thanks for the feedback. The previous wording needs a small refactor as the statement that the case property will automatically get added back to the DD on normal form submissions isn't true. How does the below sound @mkangia ?

> There are currently no cases that store data with this case property, so it is safe to delete.
Please note that if a new form or case was updated to the server recently that makes this property unsafe to delete, this will only be reflected in the Data Dictionary after a short while. This is not a blocking action and the case property can be created again if needed in the future. Please refer to the documentation for additional information about this action.

Reply:
> get added back to the DD on normal form submissions isn't true

Oh is it not!?

The updated wording sounds good to me.